### PR TITLE
ETH-699 flag closer

### PR DESCRIPTION
**Pull request by samt1803**
quick and dirty script to close expired flags, tested in dev-env and on mumbai

### CODE REVIEW

**Comment by samt1803** [comment]
There is still a problem with ts-compiling it, until now I just ran it with tsx

**Comment by hpihkala** [comment]
Works great, but for me, only after downgrading to `node-fetch@2` and installing its types.

```
npm i node-fetch@2
npm i --save-dev @types/node-fetch
```

File: packages/expired-flag-closer/src/flagCloser.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        }
+    }
+}
+
+async function main() {
+    await checkForFlags()
+    setInterval(checkForFlags, parseInt(INTERVALSEC || "900") * 1000) // default 15 minutes
+}
+
+main()
```
Comment (jtakalai) [inline packages/expired-flag-closer/src/flagCloser.ts:81]:
```suggestion
main().catch(console.error)
```

**Review by jtakalai — APPROVED** [review]

**Comment by hpihkala** [comment]
There is currently one flag which the script will not successfully close. Possibly a gas estimation issue again. The flag id is `0x823802371008f4409ed9a2a4dfc47952158d85f2-0xd7aac51f563d462b07f1441b8760bf125043bf0b-0` if you wanna have a look. The error is:

```
3|flagClos | flag id: 0x823802371008f4409ed9a2a4dfc47952158d85f2-0xd7aac51f563d462b07f1441b8760bf125043bf0b-0 failed to send tx <ref *1> Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (error={"reason":"execution reverted","code":"UNPREDICTABLE_GAS_LIMIT","method":"estimateGas","transaction":{"from":"0x73167615713A2ac178194E227B8870a560b68210","maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"maxFeePerGas":{"type":"BigNumber","hex":"0x59682f1e"},"to":"0x823802371008f4409ED9A2a4dFC47952158d85f2","data":"0xa5b83ce4000000000000000000000000d7aac51f563d462b07f1441b8760bf125043bf0bfed7cd4c28b898686b486acec881abcdeb2098d936f5a39bd4f491b41cbcfe58","type":2,"accessList":null},"error":{"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"id\":1356,\"error\":{\"code\":3,\"message\":\"execution reverted\",\"data\":\"0x4e487b710000000000000000000000000000000000000000000000000000000000000011\"}}","error":{"code":3,"data":"0x4e487b710000000000000000000000000000000000000000000000000000000000000011"},"requestBody":"{\"method\":\"eth_estimateGas\",\"params\":[{\"type\":\"0x2\",\"maxFeePerGas\":\"0x59682f1e\",\"maxPriorityFeePerGas\":\"0x59682f00\",\"from\":\"0x73167615713a2ac178194e227b8870a560b68210\",\"to\":\"0x823802371008f4409ed9a2a4dfc47952158d85f2\",\"data\":\"0xa5b83ce4000000000000000000000000d7aac51f563d462b07f1441b8760bf125043bf0bfed7cd4c28b898686b486acec881abcdeb2098d936f5a39bd4f491b41cbcfe58\"}],\"id\":1356,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://rpc-mumbai.maticvigil.com"}}, tx={"data":"0xa5b83ce4000000000000000000000000d7aac51f563d462b07f1441b8760bf125043bf0bfed7cd4c28b898686b486acec881abcdeb2098d936f5a39bd4f491b41cbcfe58","to":{},"from":"0x73167615713A2ac178194E227B8870a560b68210","type":2,"maxFeePerGas":{"type":"BigNumber","hex":"0x59682f1e"},"maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"nonce":{},"gasLimit":{},"chainId":{}}, code=UNPREDICTABLE_GAS_LIMIT, version=abstract-signer/5.7.0)
3|flagClos |     at Logger.makeError (/home/hpihkala/expired-flag-closer/node_modules/@ethersproject/logger/lib/index.js:238:21)
3|flagClos |     at Logger.throwError (/home/hpihkala/expired-flag-closer/node_modules/@ethersproject/logger/lib/index.js:247:20)
3|flagClos |     at /home/hpihkala/expired-flag-closer/node_modules/@ethersproject/abstract-signer/lib/index.js:365:47
3|flagClos |     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
3|flagClos |     at async Promise.all (index 7) {
3|flagClos |   reason: 'cannot estimate gas; transaction may fail or may require manual gas limit',
3|flagClos |   code: 'UNPREDICTABLE_GAS_LIMIT',
3|flagClos |   error: Error: cannot estimate gas; transaction may fail or may require manual gas limit [ See: https://links.ethers.org/v5-errors-UNPREDICTABLE_GAS_LIMIT ] (reason="execution reverted", method="estimateGas", transaction={"from":"0x73167615713A2ac178194E227B8870a560b68210","maxPriorityFeePerGas":{"type":"BigNumber","hex":"0x59682f00"},"maxFeePerGas":{"type":"BigNumber","hex":"0x59682f1e"},"to":"0x823802371008f4409ED9A2a4dFC47952158d85f2","data":"0xa5b83ce4000000000000000000000000d7aac51f563d462b07f1441b8760bf125043bf0bfed7cd4c28b898686b486acec881abcdeb2098d936f5a39bd4f491b41cbcfe58","type":2,"accessList":null}, error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\":\"2.0\",\"id\":1356,\"error\":{\"code\":3,\"message\":\"execution reverted\",\"data\":\"0x4e487b710000000000000000000000000000000000000000000000000000000000000011\"}}","error":{"code":3,"data":"0x4e487b710000000000000000000000000000000000000000000000000000000000000011"},"requestBody":"{\"method\":\"eth_estimateGas\",\"params\":[{\"type\":\"0x2\",\"maxFeePerGas\":\"0x59682f1e\",\"maxPriorityFeePerGas\":\"0x59682f00\",\"from\":\"0x73167615713a2ac178194e227b8870a560b68210\",\"to\":\"0x823802371008f4409ed9a2a4dfc47952158d85f2\",\"data\":\"0xa5b83ce4000000000000000000000000d7aac51f563d462b07f1441b8760bf125043bf0bfed7cd4c28b898686b486acec881abcdeb2098d936f5a39bd4f491b41cbcfe58\"}],\"id\":1356,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://rpc-mumbai.maticvigil.com"}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.7.2)
3|flagClos |       at Logger.makeError (/home/hpihkala/expired-flag-closer/node_modules/@ethersproject/logger/lib/index.js:238:21)
3|flagClos |       at Logger.throwError (/home/hpihkala/expired-flag-closer/node_modules/@ethersproject/logger/lib/index.js:247:20)
3|flagClos |       at checkError (/home/hpihkala/expired-flag-closer/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:122:20)
3|flagClos |       at JsonRpcProvider.<anonymous> (/home/hpihkala/expired-flag-closer/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:751:47)
3|flagClos |       at step (/home/hpihkala/expired-flag-closer/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:48:23)
3|flagClos |       at Object.throw (/home/hpihkala/expired-flag-closer/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:29:53)
3|flagClos |       at rejected (/home/hpihkala/expired-flag-closer/node_modules/@ethersproject/providers/lib/json-rpc-provider.js:21:65)
3|flagClos |       at process.processTicksAndRejections (node:internal/process/task_queues:95:5) {
3|flagClos |     reason: 'execution reverted',
3|flagClos |     code: 'UNPREDICTABLE_GAS_LIMIT',
3|flagClos |     method: 'estimateGas',
3|flagClos |     transaction: {
3|flagClos |       from: '0x73167615713A2ac178194E227B8870a560b68210',
3|flagClos |       maxPriorityFeePerGas: [BigNumber],
3|flagClos |       maxFeePerGas: [BigNumber],
3|flagClos |       to: '0x823802371008f4409ED9A2a4dFC47952158d85f2',
3|flagClos |       data: '0xa5b83ce4000000000000000000000000d7aac51f563d462b07f1441b8760bf125043bf0bfed7cd4c28b898686b486acec881abcdeb2098d936f5a39bd4f491b41cbcfe58',
3|flagClos |       type: 2,
3|flagClos |       accessList: null
3|flagClos |     },

```

**Comment by samt1803** [comment]
@hpihkala we are debugging the issue with the one unclosable flag and it seems to be a contract issue. We made a ticket: ETH-703

**Comment by hpihkala** [comment]
Since this turned out to be a contract issue, let's merge?

**Pull request merged**